tests/testthat/test-top-factors-getter.R

```diff
@@ -0,0 +1,89 @@
+# These tests check that the getTopFactors getter function works correctly
+# 
+# Tests include:
+# 1. Checking that the top factors order in getTopFactors matches that in 
+#    getOutDf
+# 2. Checking that the number of factors to output can be specified
+# 3. Checking that the weights are correctly ordered
+# 4. Checking that the top factors and weights are stable (always have the same
+#    values for a fixed data set and seed)
+
+context("Checking top factors getter works")
+
+set.seed(256)
+
+# Build training data
```

Comment:
I'll add some more noise to the data to deal with this (and change the tests accordingly)